test/MaterializedViewsTest.cpp

```diff
@@ -30,31 +33,79 @@ using V = Variable;
 }  // namespace
 
 // _____________________________________________________________________________
-TEST_F(MaterializedViewsTest, NotImplemented) {
-  // TODO<ullingerc> Remove this test when materialized views implementation is
-  // added. These are valid queries - the error means that they were completely
-  // parsed (and the implementation is still missing).
-  AD_EXPECT_THROW_WITH_MESSAGE(
-      qlv().parseAndPlanQuery(R"(
-        PREFIX view: <https://qlever.cs.uni-freiburg.de/materializedView/>
-        SELECT * {
-          SERVICE view:testView1 {
-            _:config view:column-s ?s ;
-                     view:column-g ?x .
-          }
+TEST_F(MaterializedViewsTest, Basic) {
+  // Write a simple view.
+  clearLog();
+  qlv().writeMaterializedView("testView1", simpleWriteQuery_);
+  EXPECT_THAT(log_.str(), ::testing::HasSubstr(
+                              "Materialized view testView1 written to disk"));
+  qlv().loadMaterializedView("testView1");
+  EXPECT_THAT(log_.str(), ::testing::HasSubstr(
+                              "Loading materialized view testView1 from disk"));
+
+  // Test index scan on materialized view.
+  std::vector<std::string> equivalentQueries{
+      R"(
+      PREFIX view: <https://qlever.cs.uni-freiburg.de/materializedView/>
+      SELECT * {
+        ?s view:testView1-g ?x .
+      }
+    )",
+      R"(
+      PREFIX view: <https://qlever.cs.uni-freiburg.de/materializedView/>
+      SELECT * {
+        SERVICE view:testView1 {
+          _:config view:column-s ?s ;
+                   view:column-g ?x .
         }
-      )"),
-      ::testing::HasSubstr(
-          "Materialized views are currently not supported yet"));
-  AD_EXPECT_THROW_WITH_MESSAGE(
-      qlv().parseAndPlanQuery(R"(
-        PREFIX view: <https://qlever.cs.uni-freiburg.de/materializedView/>
+      }
+    )",
+      // Regression test (subquery).
+      R"(
+      PREFIX view: <https://qlever.cs.uni-freiburg.de/materializedView/>
+      SELECT * {
         SELECT * {
           ?s view:testView1-g ?x .
         }
-      )"),
+      }
+    )",
+  };
+  for (const auto& query : equivalentQueries) {
+    auto [qet, qec, parsed] = qlv().parseAndPlanQuery(query);
+    auto res = qet->getResult(false);
+
+    EXPECT_THAT(qet->getRootOperation()->getCacheKey(),
+                ::testing::HasSubstr("testView1"));
+
+    EXPECT_EQ(res->idTable().numRows(), 4);
+    auto col = qet->getVariableColumn(Variable{"?x"});
+    for (size_t i = 0; i < res->idTable().numRows(); ++i) {
+      auto id = res->idTable().at(i, col);
+      EXPECT_EQ(id.getDatatype(), Datatype::Int);
+      EXPECT_EQ(id.getInt(), 1);
+    }
+  }
```

Comment:
We have some matchers for IdTables, set up the full expected result in your infrastructure (near the query that you write etc.), and then do EXPECT_THAT(res->idTable(), matchesIdTable(expectedFullResult) etc.